sources/packages/backend/apps/api/src/route-controllers/student/student.students.controller.e2e-spec.ts

```diff
@@ -1,113 +1,63 @@
+    accessToken = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
     );
-    accesstoken = token.access_token;
-    const moduleFixture: TestingModule = await Test.createTestingModule({
-      imports: [
-        DatabaseModule,
-        AuthModule,
-        AppStudentsModule,
-        NotificationsModule,
-        createMockedZeebeModule(),
-        MockedQueueModule,
-      ],
-    }).compile();
-    userService = await moduleFixture.get(UserService);
+    const { nestApplication, module, dataSource } =
+      await createTestingAppModule();
+    const moduleFixture = module;
     atbcService = await moduleFixture.get(ATBCService);
     studentService = await moduleFixture.get(StudentService);
-    sinValidationService = await moduleFixture.get(SINValidationService);
-    app = moduleFixture.createNestApplication();
-    await app.init();
+    app = nestApplication;
+    appDataSource = dataSource;
   });
 
-  it("should return an HTTP 200 status when applying for PD and student is valid", async () => {
-    // Create fake student in SIMS DB
-    const fakeStudent = new Student();
-    fakeStudent.sinConsent = true;
-    fakeStudent.birthDate = faker.date.past(18).toISOString();
-    fakeStudent.gender = "F";
-    fakeStudent.contactInfo = {
-      address: {
-        addressLine1: faker.address.streetAddress(),
-        city: faker.address.city(),
-        country: "Canada",
-        provinceState: "ON",
-        postalCode: faker.address.zipCode(),
-      },
-      phone: faker.phone.phoneNumber(),
-    };
-    const simsUser = new User();
-    simsUser.userName = process.env.E2E_TEST_STUDENT_USERNAME;
-    simsUser.email = faker.internet.email();
-    simsUser.firstName = faker.name.firstName();
-    simsUser.lastName = faker.name.lastName();
-    fakeStudent.user = simsUser;
-
-    // Save the student in SIMS.
-    await studentService.save(fakeStudent);
-
-    // creating mockup for ATBCCreateClient, this function actually calls the ATBC server to create the student profile
+  it("Should return an HTTP 200 status when applying for PD and student is valid", async () => {
+    // Arrange
+    // Creating mockup for ATBCCreateClient.
+    // This function actually calls the ATBC server to create the student profile.
     jest.spyOn(atbcService, "createClient").mockImplementation(async () => {
       return {} as ATBCCreateClientResponse;
     });
 
-    const sinValidation = new SINValidation();
-    sinValidation.student = fakeStudent;
-    sinValidation.isValidSIN = true;
-    fakeStudent.sinValidation = sinValidation;
-    sinValidation.sin = "706941291";
-
+    // Act/Assert
+    await request(app.getHttpServer())
+      .patch("/students/atbc/apply-pd-status")
+      .auth(accessToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK);
+  });
+  afterAll(async () => {
+    // Putting the student in the same state as before the test.
+    const fakeStudent = await getStudentByFakeStudentUserType(
```

Comment:
I am not following this code here. Is the `studentPDSentAt` changed by these tests?

Reply:
yes, when the student apply for PD status it stamps the datetime in `pd_date_sent` column and if that is not reverted, next time the test is run it will get a 422 "Unprocessable Entity".
![image](https://user-images.githubusercontent.com/78114138/225753141-62a2e13f-84f1-4e63-8354-4029a012a204.png)

Reply:
A see the point but:
1. It happens only on the DEV machine while DEV is reusing the same DB over and over again. The tests are intended to be executed on a clean DB.
2. This is the type of test where we are changing the student data in a way that can potentially cause side effects in other tests. Should we maybe not be using the `FakeStudentUserType1` then?

Reply:
> 1. It happens only on the DEV machine while DEV is reusing the same DB over and over again. The tests are intended to be executed on a clean DB.
I thought that the idea was to run the tests as many times as we needed with the same result but ok, I got what you say. 

> 2. This is the type of test where we are changing the student data in a way that can potentially cause side effects in other tests. Should we maybe not be using the `FakeStudentUserType1` then?
Yes, but that other user should be used specifically for this test then, right?

Reply:
Ok, we discussed and decided that I'm going to use the same user for both tests as they don't affect one another. Thanks!